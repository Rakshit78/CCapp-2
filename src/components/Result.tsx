import { Button, Container, Typography } from '@mui/material';
import { PieChart } from 'react-minimal-pie-chart';
type QuizProps = {
  navigate: any;
  que: any;
  username: string;
  score: number;
};
const Result: React.FC<QuizProps> = ({ que, navigate, username, score }) => {
  return (
    <Container
      style={{
        width: '100vw',
        height: '100vh',
        display: 'grid',
        placeItems: 'center',
      }}
    >
      <Container
        maxWidth='sm'
        style={{ background: '#a5a5a5', borderRadius: '14px', padding: '20px' }}
      >
        <Button
          onClick={() => {
            navigate('/');
            window.location.reload();
          }}
          role='retake'
        >
          Retake
        </Button>
        <Typography variant='h4'>
          Hey!! {username} you got {(score / 5) * 100}%
        </Typography>
        <PieChart
          data-testid='pie-chart'
          style={{ width: '100px' }}
          data={[
            { title: 'One', value: score, color: '#E38627' },
            { title: 'Two', value: 5 - score, color: '#C13C37' },
          ]}
        />
        {que.map((res: any) => {
          return (
            <>
              <Typography variant='body1'>{res.q}</Typography>
              <Typography variant='body1'>
                <b> Correct Answer:</b>
                {JSON.stringify(res.a)}
              </Typography>
              <Typography variant='body1'>
                <b>Your Answer:</b>
                {JSON.stringify(res.usera)}
              </Typography>
            </>
          );
        })}
      </Container>
    </Container>
  );
};

export default Result;
