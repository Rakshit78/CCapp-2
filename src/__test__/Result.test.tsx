import React from 'react';
import { render, screen } from '@testing-library/react';
import Result from '../components/Result';

describe('Result test', () => {
  const que: any = [];
  const username: string = '';
  const score: number = 0;
  test('Result renders without crash', () => {
    render(
      <Result
        navigate={undefined}
        que={que}
        username={username}
        score={score}
      />
    );
  });

  test('Result renders without crash', () => {
    const component = render(
      <Result
        navigate={undefined}
        que={que}
        username={username}
        score={score}
      />
    );
    expect(component).toMatchSnapshot();
  });

  test('Result renders without crash', () => {
    render(
      <Result
        navigate={undefined}
        que={que}
        username={username}
        score={score}
      />
    );
    expect(screen.getByRole('retake')).toBeInTheDocument();
  });
});
