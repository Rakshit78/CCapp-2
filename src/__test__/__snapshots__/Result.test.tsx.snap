// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result test Result renders without crash 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
        style="width: 100vw; height: 100vh; display: grid; place-items: center;"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthSm css-cuefkz-MuiContainer-root"
          style="background: rgb(165, 165, 165); border-radius: 14px; padding: 20px;"
        >
          <button
            class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
            role="retake"
            tabindex="0"
            type="button"
          >
            Retake
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <h4
            class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
          >
            Hey!! 
             you got 
            0
            %
          </h4>
          <svg
            height="100%"
            style="width: 100px;"
            viewBox="0 0 100 100"
            width="100%"
          >
            <path
              d=""
              fill="none"
              stroke="#E38627"
              stroke-width="50"
            >
              <title>
                One
              </title>
            </path>
            <path
              d="M 75 50 A 25 25 0 1 1 74.99999999619229 49.99956366768703"
              fill="none"
              stroke="#C13C37"
              stroke-width="50"
            >
              <title>
                Two
              </title>
            </path>
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
      style="width: 100vw; height: 100vh; display: grid; place-items: center;"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthSm css-cuefkz-MuiContainer-root"
        style="background: rgb(165, 165, 165); border-radius: 14px; padding: 20px;"
      >
        <button
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          role="retake"
          tabindex="0"
          type="button"
        >
          Retake
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <h4
          class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
        >
          Hey!! 
           you got 
          0
          %
        </h4>
        <svg
          height="100%"
          style="width: 100px;"
          viewBox="0 0 100 100"
          width="100%"
        >
          <path
            d=""
            fill="none"
            stroke="#E38627"
            stroke-width="50"
          >
            <title>
              One
            </title>
          </path>
          <path
            d="M 75 50 A 25 25 0 1 1 74.99999999619229 49.99956366768703"
            fill="none"
            stroke="#C13C37"
            stroke-width="50"
          >
            <title>
              Two
            </title>
          </path>
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
